### PR TITLE
tau-hydrogen: init at 1.0.11

### PR DESCRIPTION
**Pull request by Ashvith10**
## Description of changes

Adds tau-hydrogen icon theme.

## Things done

- Added @Ashvith10 as the maintainer
- Added the tau-hydrogen icon theme

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Lord-Valen** [comment]
`tau-hydrogen` isn't taken, so you could probably just use that pname.

**Comment by Ashvith10** [comment]
@Lord-Valen should I also change the directory where the package resides?

**Comment by Lord-Valen** [comment]
> @Lord-Valen should I also change the directory where the package resides?

I would recommend doing so for consistency.

**Review by Lord-Valen — APPROVED** [review]
LGTM

**Comment by Lord-Valen** [comment]
One thing I often forget to mention: Can you use the finalAttrs pattern for the derivation?

**Comment by Ashvith10** [comment]
@Lord-Valen I'm going to be with a laptop for a week or two. Could you add those changes? I think I've enabled the setting to let maintainers edit PR.

**Comment by Ashvith10** [comment]
@Lord-Valen can you verify if the new changes are acceptable?

File: pkgs/data/icons/tau-hydrogen/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+stdenv.mkDerivation (finalAttrs: {
+  pname = "tau-hydrogen";
+  version = "1.0.11";
+
+  src = fetchFromGitHub {
+    owner = "tau-OS";
+    repo = finalAttrs.pname;
+    rev = finalAttrs.version;
+    sha256 = "sha256-ECrRWWS/Am0lfCIJw/BVZg53oLw79Im8d8KgAYxE+pw=";
+  };
```
Comment (Lord-Valen) [inline pkgs/data/icons/tau-hydrogen/default.nix:21]:
```suggestion
  src = fetchFromGitHub {
    owner = "tau-OS";
    repo = "tau-hydrogen";
    rev = finalAttrs.version;
    hash = "sha256-ECrRWWS/Am0lfCIJw/BVZg53oLw79Im8d8KgAYxE+pw=";
  };
```

**Comment by Lord-Valen** [comment]
Squash to two commits.

**Comment by Ashvith10** [comment]
@Lord-Valen how about now? Is it okay?

**Review by Lord-Valen — APPROVED** [review]

**Review by lilyinstarlight — COMMENTED** [review]
I have two nits, but otherwise this looks good! Thank you for contributing to nixpkgs :)

Also I just noticed this, but the commit message should be `tau-hydrogen: init at 1.0.11`, to match the attr (I assume it got renamed in earlier reviews) so that ofborg automatically builds it properly

**Review by lilyinstarlight — APPROVED** [review]
Looks good to me now! Congrats on your first package in nixpkgs :)

**Pull request merged**